src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -0,0 +1,751 @@
+        }
+    }
+
+    protected void initViews(List<View> views) throws IOException {
+        AllView v = new AllView("All", this);
+        views.add(v);
+        v.save();
+    }
+
+    @Override
+    public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOException {
+        super.onLoad(parent, name);
+        init();
+        final Thread t = Thread.currentThread();
+        String n = t.getName();
+        try {
+            if (items == null) {
+                // When Jenkins is getting reloaded, we want children being loaded to be able to find existing items that they will be overriding.
+                // This is necessary for them to correctly keep the running builds, for example.
+                // ItemGroupMixIn.loadChildren handles the rest of this logic.
+                Item current = parent.getItem(name);
+                if (current != null && current.getClass() == getClass()) {
+                    this.items = ((AbstractFolder) current).items;
+                }
+            }
+
+            items = loadChildren(this, getJobsDir(), new Function1<String,I>() {
+                @Override
+                public String call(I item) {
+                    String fullName = item.getFullName();
+                    t.setName("Loading job " + fullName);
+                    float percentage = 100.0f * jobEncountered.incrementAndGet() / Math.max(1, jobTotal.get());
+                    long now = System.currentTimeMillis();
+                    if (loadingTick == 0) {
+                        loadingTick = now;
+                    } else if (now - loadingTick > TICK_INTERVAL) {
+                        LOGGER.log(Level.INFO, String.format("Loading job %s (%.1f%%)", fullName, percentage));
+                        loadingTick = now;
+                    }
+                    // TODO loadChildren does not support decoding folder names
+                    return item.getName();
+                }
+            });
+        } finally {
+            t.setName(n);
+        }
+    }
+
+    @Override
+    public AbstractFolderDescriptor getDescriptor() {
+        // Seems to work even though AbstractFolder is not Describable, hmm
+        return (AbstractFolderDescriptor) Jenkins.getActiveInstance().getDescriptorOrDie(getClass());
+    }
+
+    /**
+     * May be used to enumerate or remove properties.
+     * To add properties, use {@link #addProperty}.
+     */
+    public DescribableList<AbstractFolderProperty<?>,AbstractFolderPropertyDescriptor> getProperties() {
+        return properties;
+    }
+
+    @SuppressWarnings("rawtypes") // else setOwner will not compile
+    public void addProperty(AbstractFolderProperty p) throws IOException {
+        if (!p.getDescriptor().isApplicable(getClass())) {
+            throw new IllegalArgumentException(p.getClass().getName() + " cannot be applied to " + getClass().getName());
```

Comment:
Note that this is stricter than the original version in `Folder`, which is a potential compatibility issue.